schema/schema_test.go

```diff
@@ -98,7 +101,82 @@ func TestCreateCollectionFromSchema(t *testing.T) {
 		},
 		"primary_key": ["K1"]
 	}`)
-		_, err := CreateCollection("d1", "t1", schema)
+		_, err := Build("t1", schema)
 		require.Equal(t, "unsupported primary key type detected 'number'", err.(*api.TigrisDBError).Error())
 	})
+	t.Run("test_complex_types", func(t *testing.T) {
+		schema := []byte(`{
+	"name": "t1",
+	"properties": {
+		"id": {
+			"type": "integer"
+		},
+		"random": {
+			"type": "string",
+			"contentEncoding": "base64",
+			"maxLength": 1024
+		},
+		"product": {
+			"type": "string",
+			"maxLength": 100
+		},
+		"price": {
+			"type": "number"
+		},
+		"simple_items": {
+			"type": "array",
+			"items": {
+				"type": "integer"
+			}
+		},
+		"simple_object": {
+			"type": "object",
+			"properties": {
+				"name": { "type": "string" }
+			}
+		},
+		"product_items": {
+			"type": "array",
+			"items": {
```

Comment:
Is it checked that if type is arrays then items must be set. Same if type is object then properties must be set.
Optionally we can check if properties and items is set for other types and reject, to help finding programming errors.

Reply:
good point, added more guards.